mlpp_lib/imputer.py

```diff
@@ -0,0 +1,287 @@
+        first_key = next(iter(in_dict))
+
+        #TODO: this part should ensure retro-compatibility and therefore access DataTransformer ...
+        if first_key not in [
+            cls.__name__ for cls in DataImputation.__subclasses__()
+        ]:
+            channels = list(in_dict["mean"]["data_vars"].keys())
+            init_dict = {"method_vars_dict": {"Standardizer": channels}}
+            method_dicts = {"Standardizer": in_dict}
+        else:
+            method_vars_dict = {}
+            method_dicts = {}
+            for method_name, inner_dict in in_dict.items():
+                method_vars_dict[method_name] = inner_dict.pop("channels")
+                method_dicts[method_name] = inner_dict
+            init_dict = {"method_vars_dict": method_vars_dict}
+            method_dicts = in_dict
+        return init_dict, method_dicts
+
+    def to_dict(self):
+        """Convert the DataImputer instance to a dictionary."""
+        out_dict = {}
+        for name, (imputer, variables) in self.imputers.items():
+            out_dict_tmp = imputer.to_dict()
+            out_dict_tmp["channels"] = variables
+            out_dict[name] = out_dict_tmp
+        return out_dict
+
+    @classmethod
+    def from_json(cls, in_fn: str) -> Self:
+        """Create a DataImputer instance from a JSON file."""
+        with open(in_fn, "r") as f:
+            in_dict = json.load(f)
+        return cls.from_dict(in_dict)
+
+    def save_json(self, out_fn: str) -> None:
+        """Save the DataImputer configuration to a JSON file."""
+        out_dict = self.to_dict()
+        if not out_dict or not out_dict[list(out_dict.keys())[0]]:
+            raise ValueError(f"{self.name} wasn't fit to data")
+        with open(out_fn, "w") as outfile:
+            json.dump(out_dict, outfile, indent=4)
+
+class DataImputation:
+    """
+    Abstract class for nromalization techniques in a xarray.Dataset object.
+    """
+    
+    @abstractmethod
+    def fit(self, dataset: xr.Dataset, dims: Optional[list] = None) -> None:
+        pass
+
+    @abstractmethod
+    def fill(self, dataset: xr.Dataset, dims: Optional[list] = None) -> tuple[xr.Dataset, ...]:
+        pass
+
+    @abstractmethod
+    def inverse_fill(self, *datasets: xr.Dataset) -> tuple[xr.Dataset, ...]:
+        pass
+
+    @abstractmethod
+    def from_dict(self, in_dict) -> Self:
+        pass
+
+    @abstractmethod
+    def to_dict(self) -> dict:
+        pass
+
+
+@dataclass
+class ConstantImputer(DataImputation):
+    """
+    Fill missing values with a constant value.
+    """
+    #TODO: rethink a bit the logic, a priori fillvalue could have one distinct value for each variable
+
+    fillvalue: dict[str, float] = field(init=True, default=-5)
+    name = "ConstantImputer"
+
+    def fit(self,
+        dataset: xr.Dataset,
+        dims: Optional[list] = None,
+    ):
+        self.fillvalue = self.fillvalue
```

Comment:
Here we currently don't ask for all variables to be filled. This is not an issue as we remove NaNs afterwards anyway, but this behaviour may not be expected by users. 
Should we force for all vars to be present ? Or have a "default" value assigned to the rest of variables ?